digitalforms-api/src/main/java/ca/bc/gov/open/pssg/rsbc/digitalforms/controller/QueryServiceController.java

```diff
@@ -61,11 +66,27 @@ public QueryServiceController(QueryServiceImpl irpService) {
 		if (ordsResp.getRespCode() == DigitalFormsConstants.ORDS_SUCCESS_CD) {	
 			JSONResponse<ProhibitionStatusResponse> r = new JSONResponse<>(resp);
 			return new ResponseEntity<>(r, HttpStatus.OK);
+			
 		} else if (ordsResp.getRespCode() == DigitalFormsConstants.ORDS_FAILURE_CD) {
+			
 			JSONResponse<ProhibitionStatusResponse> r = new JSONResponse<>(null); 
-		    r.setError(new JSONError("Notice number not found", HttpStatus.NOT_FOUND.value()));
-		    return new ResponseEntity<>(r, HttpStatus.NOT_FOUND);
+			
+			// parse the error response. 
+			ObjectMapper mapper = new ObjectMapper();
+			ProhibitionStatusErrorResponse error = null; 
+			try {
+				error = mapper.readValue(ordsResp.getRespMsg(), ProhibitionStatusErrorResponse.class);
+			} catch (JsonMappingException e) {
+				e.printStackTrace();
+			} catch (JsonProcessingException e) {
+				e.printStackTrace();
+			}
+			
+			r.setError(new JSONError(error.getStatusMessage(), HttpStatus.NOT_FOUND.value()));
+			return new ResponseEntity<>(r, HttpStatus.NOT_FOUND);
+		   
```

Comment:
Just a thought, if these exceptions are expected to be thrown by other controllers, we can configure it in the @ControllerAdvice class so that we dont have to handle each of them separately.

Reply:
Fair enought. This can be done tomorrow or later. Maybe you can explain this to me. I just had to get it working well enough for the demo for now. Thanks